core/build_playerglobal/src/lib.rs

```diff
@@ -79,27 +79,12 @@ pub fn build_playerglobal(
     Ok(())
 }
 
-// Resolve the 'name' field of a `Multiname`. This only handles the cases
-// that we need for our custom `playerglobal.swf` (
-fn resolve_multiname_name<'a>(abc: &'a AbcFile, multiname: &Multiname) -> &'a str {
-    if let Multiname::QName { name, .. } | Multiname::Multiname { name, .. } = multiname {
-        &abc.constant_pool.strings[name.0 as usize - 1]
-    } else {
-        panic!("Unexpected Multiname {:?}", multiname);
-    }
-}
-
-// Like `resolve_multiname_name`, but for namespaces instead.
-fn resolve_multiname_ns<'a>(abc: &'a AbcFile, multiname: &Multiname) -> &'a str {
-    if let Multiname::QName { namespace, .. } = multiname {
-        let ns = &abc.constant_pool.namespaces[namespace.0 as usize - 1];
-        if let Namespace::Package(p) = ns {
-            &abc.constant_pool.strings[p.0 as usize - 1]
-        } else {
-            panic!("Unexpected Namespace {:?}", ns);
-        }
-    } else {
-        panic!("Unexpected Multiname {:?}", multiname);
+/// Retrieve a string from the constant pool.
+/// Index zero is the empty string.
+fn pool_string(abc: &AbcFile, index: Index<String>) -> &str {
+    match index.0.checked_sub(1) {
+        Some(i) => &abc.constant_pool.strings[i as usize],
+        None => "",
     }
 }
```

Comment:
I think this would probably be useful as method on `AbcFile`, such as `AbcFile::get_string`. Could do the other constant pool types too, maybe as a future PR.

Reply:
I had the same thought. A future PR seems a better fit since currently there are tons of direct constant pool accesses across the codebase.